pkg/etcdutil/etcdutil.go

```diff
@@ -254,7 +263,7 @@ func GetEtcdEndPointsSorted(ctx context.Context, clientMaintenance client.Mainte
 }
 
 // TakeAndSaveFullSnapshot takes full snapshot and save it to store
-func TakeAndSaveFullSnapshot(ctx context.Context, client client.MaintenanceCloser, store brtypes.SnapStore, lastRevision int64, cc *compressor.CompressionConfig, suffix string, isFinal bool, logger *logrus.Entry) (*brtypes.Snapshot, error) {
+func TakeAndSaveFullSnapshot(ctx context.Context, client client.MaintenanceCloser, store brtypes.SnapStore, tempDir string, lastRevision int64, cc *compressor.CompressionConfig, suffix string, isFinal bool, logger *logrus.Entry) (*brtypes.Snapshot, error) {
```

Comment:
I would not like to change the function `TakeAndSaveFullSnapshot()` as this is not a refactoring PR. 
Feel free to open a separate issue if you want to refactor `TakeAndSaveFullSnapshot()`.

Reply:
Since you are touching this function and adding more 15+ lines, I don't want to grow this function. This won't help readability of code. Thats why I suggest to move the code into smaller functions, each function name clearly conveys the functionality

Reply:
make sense, but I would not like to do that in the same PR for following reasons:
1. That will change the scope of PR from adding feature PR to refactoring PR which will be hard to track after 1year (say) if some bug is found in some part of code.
2. With this refactoring change, it will create a huge overhead of manual testing, performance testing etc like taking snapshots, compression of snapshots, restoration of compressed and uncompressed snapshots + verification of snapshots (feature is being added) all needs to be fully tested.
3. It will completely change the scope of the PR, which will also invalidate the `LGTM'ed` of reviewers those who has already reviewed the PR as it will become a new PR.

That's I don't want to do this in the same PR.

Reply:
**Point 1:** Do you see this as a major refactor? It’s essentially just restructuring a few existing lines into two functions for better readability.

**Point 2:** This point isn’t entirely valid, as we are already planning to test this functionality as part of the release process. This includes verifying the new integrity checks that have been added.

**Point 3:** Whenever we introduce new changes or code, it’s common to make minor adjustments to the surrounding logic for better structure. In this case, it’s simply wrapping the logic into smaller, reusable functions.

That said, this isn’t a significant code change. Let’s get some thoughts from other reviewers as well. 

WDYT, @anveshreddy18 and @renormalize?

Reply:
>Do you see this as a major refactor? I

yes, I do because of point 2
I have mentioned the my reasoning in this comment https://github.com/gardener/etcd-backup-restore/pull/779#discussion_r1865254302, I'm not making this PR as refactoring PR. Please feel free to open a separate issue later if somebody have time they can pick it up.

Reply:
> This point isn’t entirely valid, as we are already planning to test this functionality as part of the release process. This includes verifying the new integrity checks that have been added.

point 2 is totally valid, Do currently we have e2e tests which will do what I have mentioned in point 2 or not ? the answer is no, then it will be create a huge overhead. 

> we are already planning to test this functionality as part of the release process.

Currently in release process nobody do the performance testing ... question is not about future, question is about do we have it now or not ? I don't think so, we have it snf I don't see it coming in near by weeks.

Reply:
It seems your concern is about not adding the functions `createEtcdSnapshot`, `saveSnapshotToStore`, and `compressSnapshotDataIfNeeded` to encapsulate the existing logic. 

I’m okay with that approach. Are you aligned with the other review comments? Let me know if there’s anything else to discuss!